kernel/init.c

```diff
@@ -105,6 +105,33 @@ k_tid_t const _idle_thread = (k_tid_t)&_idle_thread_s;
 #endif
 K_THREAD_STACK_DEFINE(_interrupt_stack, CONFIG_ISR_STACK_SIZE);
 
+/*
+ * Similar idle thread & interrupt stack definitions for the
+ * auxilliary CPUs.  The declaration macros aren't set up to define an
+ * array, so do it with a simple test for up to 4 processors.  Should
+ * clean this up in the future.
+ */
```

Comment:
Consider:

K_THREAD_STACK_ARRAY_DEFINE(idle_stacks, CONFIG_MP_NUM_CPUS, IDLE_STACK_SIZE);

k_tid_t const _idle_threads[CONFIG_MP_NUM_CPUS];

and so forth